src/test/javascript/spec/component/legal/data-export-confirmation-dialog.component.spec.ts

```diff
@@ -55,26 +59,32 @@ describe('DataExportConfirmationDialogComponent', () => {
         expect(inputFormGroup).not.toBeNull();
     });
 
-    it('should correctly enable and disable request button', () => {
+    it('should correctly enable and disable request button', async () => {
         // Form can't be submitted if the expected login doesn't match the entered login
         comp.expectedLogin = 'login';
         comp.enteredLogin = '';
         comp.dialogError = new Observable<string>();
         fixture.detectChanges();
-        let submitButton = debugElement.query(By.css('.btn.btn-primary'));
-        expect(submitButton.nativeElement.disabled).toBeTrue();
+        await fixture.whenStable();
+        expect(comp.dataExportConfirmationForm.invalid).toBeTrue();
+        // TODO: why do changes not propagate to the UI?
```

Comment:
Does sb. know why the form is `invalid`, but `disabled` is not `true` on the button? Somehow, it does not seem to render, but I don't know why.

Reply:
we usually use `fakeAsync`